prow/cmd/branchprotector/protect.go

```diff
@@ -288,42 +250,35 @@ func (p *Protector) UpdateRepo(orgName string, repo string, repoDefaults config.
 	}
 
 	for _, branchName := range branches {
-		if err := p.UpdateBranch(orgName, repo, branchName, repoDefaults.Branches[branchName], protect, rc, rp); err != nil {
+		if err := p.UpdateBranch(orgName, repo, branchName, repoDefaults.Branches[branchName], protect); err != nil {
 			return fmt.Errorf("UpdateBranch(%s, %s, %s, ...) failed: %v", orgName, repo, branchName, err)
 		}
 	}
 	return nil
 }
 
 // Update the branch with the specified configuration
-func (p *Protector) UpdateBranch(orgName, repo string, branchName string, branchDefaults config.Branch, protect *bool, contexts, pushers []string) error {
-	bc := append([]string{}, contexts...)
-	bpush := append([]string{}, pushers...)
+func (p *Protector) UpdateBranch(orgName, repo string, branchName string, branchDefaults config.Branch, protect *bool) error {
 	if branchDefaults.Protect != nil {
 		protect = branchDefaults.Protect
 	}
-	bc = append(bc, branchDefaults.Contexts...)
-	bpush = append(bpush, branchDefaults.Pushers...)
 
-	if protect == nil {
-		return errors.New("protect should not be nil")
+	bp, err := p.cfg.GetBranchProtection(orgName, repo, branchName)
+	if err != nil {
+		log.Printf("unable to compute requirement for %s/%s@%s", orgName, repo, branchName)
```

Comment:
Are we using `@` anywhere else to separate repo from branch? I know `bootstrap.py` expects `=`

Reply:
no. Just in this log statement

Reply:
Github uses `fejta:branch`, bootstrap uses `--repo=fejta/repo=branch` please use one of those two forms

Reply:
done